src/main/java/trainingportal/controller/CourseController.java

```diff
@@ -1,85 +1,89 @@
 package trainingportal.controller;
 
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.validation.BindingResult;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.servlet.ModelAndView;
 import org.springframework.web.servlet.mvc.support.RedirectAttributes;
-import trainingportal.dao.CourseDAOImpl;
 import trainingportal.model.Course;
+import trainingportal.service.CourseServiceImpl;
 
 import java.util.List;
 
 @Controller
 public class CourseController {
-    @Autowired
-    private CourseDAOImpl courseDAO;
-
-    @RequestMapping(value = "/course_create", method = RequestMethod.GET)
-    public ModelAndView showAllCourses(ModelAndView model, Long Id) {
-//    public String showAllCourses(Model model) {
-        List<Course> course = courseDAO.getCoursAll();
-//        model.addAttribute("courseAll", list);
-        model.addObject("courseAll", course);
-        model.setViewName("CourseCreator/course_create");
-        return model;
+
+    private final CourseServiceImpl courseService;
+
+    public CourseController(CourseServiceImpl courseService) {
+        this.courseService = courseService;
+    }
+    //used fo old realisation
+//    @Autowired
+//    CourseDAOImpl courseDAO;
+
+    @RequestMapping(value = "/course_create")
+    public ModelAndView showCoursesList(Long courseId, ModelAndView modelAndView) {
+        List<Course> courseList = courseService.CoursesList();
+//        List<Course> courseList = courseService.getAllCoursesById(courseId);
+//        Course courseList = courseService.findCourseById(courseId);
+        modelAndView.addObject("courseList", courseList);
+        modelAndView.setViewName("courseCreator/course_create");
+        return modelAndView;
     }
 
     @RequestMapping("/course-add")
     public ModelAndView addCourse(ModelAndView model) {
 
-        model.addObject("courseAdd", new Course());
-        model.setViewName("CourseCreator/course_add");
+        model.addObject("course", new Course());
+        model.setViewName("courseCreator/course_add");
 
         return model;
     }
 
-    @RequestMapping(value = "saveCourse", method = RequestMethod.POST)
-    public ModelAndView saveCourse(Course course, ModelAndView model, RedirectAttributes redir) {
+    @RequestMapping(value = "courseSave", method = RequestMethod.POST)
+    public ModelAndView saveCourse(Course course, ModelAndView model) {
 
-        courseDAO.save(course, course.getId());
-        model.setViewName("CourseCreator/course_add");
+        courseService.saveCourse(course);
+//        courseDAO.save(course);
+        model.setViewName("courseCreator/course_add");
```

Comment:
You can use 
```
modelAndView.setViewName("redirect:/course_create");
```
instead of
```
model.setViewName("courseCreator/course_add");
```

Reply:
used still dont help

model.setViewName("redirect:/course_create"); 
used this couse  public ModelAndView saveCourse(Course course, ModelAndView model) {